common:
    lid_topic:  "/livox/lidar"
    imu_topic:  "/livox/imu"
    camera_topic: "/usb_cam/image_raw"
    time_sync_en: false         # ONLY turn on when external time synchronization is really not possible

    rootDir: "/home/yixin-f/fast-lio2/src/data_lio/"
    savePCD: true                         # https://github.com/TixiaoShan/LIO-SAM/issues/3
    savePCDDirectory: "PCDs/"        
    saveSCD: true 
    saveSCDDirectory: "SCDs/"
    saveLOG: true
    saveLOGDirectory: "LOG/"
    map_save_en: true  # global map
    pcd_save_interval: 1                 # how many LiDAR frames saved in each pcd
    
preprocess:
    lidar_type: 1                # 1 for Livox serials LiDAR, 2 for Velodyne LiDAR
    livox_type: 1                # 1 for LIVOX_CUS
    scan_line:  4
    blind: 0.5
    feature_enabled: 0  # bool
    point_filter_num: 3 # bool
    scan_rate: 10 
    time_unit: 1

camera:
    camera_en: false
    camera_external: [ 0.000000, -1.000000, 0.000000, 0.000000,
                       0.000000, 0.000000, -1.000000, 0.300000,
                       1.000000, 0.000000, 0.000000, 0.000000,
                       0.000000, 0.000000, 0.000000, 1.000000]        #相机2雷达外参
    camera_internal: [516.235840, 0.000000, 641.125888, 0.000000,
                      0.000000, 533.185608, 338.725370, 0.000000,
                      0.000000, 0.000000, 1.000000, 0.000000]         #相机内参

mapping:
    acc_cov: 0.1
    gyr_cov: 0.1
    b_acc_cov: 0.0001
    b_gyr_cov: 0.0001
    fov_degree:    360
    det_range:     100.0
    cube_len: 1000  
    extrinsic_est_en:  true      # true: enable the online estimation of IMU-LiDAR extrinsic
    mappingSurfLeafSize: 0.2
    keyframeAddingDistThreshold: 1.0
    keyframeAddingAngleThreshold: 0.2
    extrinsic_T: [ 0.05512, 0.02226, -0.0297 ]
    extrinsic_R: [ 1, 0, 0,
                   0, 1, 0,
                   0, 0, 1]

ikdtree:
    max_iteration: 4
    recontructKdTree: true  # if path is too large, the rviz will crash, but default is false
    kd_step: 40
    filter_size_map_min: 0.2

loop:
    loopClosureEnableFlag: false
    loopClosureFrequency: 1.0
    historyKeyframeSearchRadius: 10.0
    historyKeyframeSearchTimeDiff: 30.0
    historyKeyframeSearchNum: 1  # submap
    historyKeyframeFitnessScore: 0.2

segment:
    ground_en: false
    tollerance_en: false  # // TODO: error !
    sensor_height: 1.5
    z_tollerance: 2.0
    rotation_tollerance: 0.2

publish:
    path_en:  true
    scan_publish_en:  true       # false: close all the point cloud output
    dense_publish_en: false       # false: low down the points number in a global-frame point clouds scan.
    scan_bodyframe_pub_en: true  # true: output the point cloud scans in IMU-body-frame

visualization:
    globalMapVisualizationSearchRadius: 10.0  # ikdtree search
    globalMapVisualizationPoseDensity: 1   # ikdtree pose
    globalMapVisualizationLeafSize: 0.2  # ikdtree point
    visulize_IkdtreeMap: true
